### PR TITLE
feat(android-report): created/re-factored header section for unified report 

### PR DESCRIPTION
**Pull request by AhmedAbdoOrtiga**
#### Description of changes
- Added a new header section for unified report.
- Updated HeaderSection in ReportSectionFactory to: 
  `HeaderSection: ReactFCWithDisplayName<SectionProps> | ReactFCWithDisplayName;`

**Web:**
image

**Unified**
image

#### Pull request checklist

- [x] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/reports/components/report-sections/report-section-factory.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ export type ReportSectionFactory = {
     HeadSection: ReactFCWithDisplayName;
     BodySection: ReactFCWithDisplayName;
     ContentContainer: ReactFCWithDisplayName;
-    HeaderSection: ReactFCWithDisplayName<SectionProps>;
+    HeaderSection: ReactFCWithDisplayName<SectionProps> | ReactFCWithDisplayName;
```
Comment (waabid) [inline src/reports/components/report-sections/report-section-factory.tsx:39]: Is this change really necessary? As in, do the typings break if that's not the case?

Comment (waabid) [reply src/reports/components/report-sections/report-section-factory.tsx:39]: I'm just thinking an empty set of props should be a subset of SectionProps so as a result, this should just work, no?

Comment (AhmedAbdoOrtiga) [reply src/reports/components/report-sections/report-section-factory.tsx]: actually it's not necessary

File: src/electron/views/report/unified-header-section.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// Licensed under the MIT License.
+import { NamedFC } from 'common/react/named-fc';
+import { androidAppTitle } from 'content/strings/application';
+import { BrandWhite } from 'icons/brand/white/brand-white';
+import * as React from 'react';
+
+export const UnifiedHeaderSection = NamedFC('UnifiedHeaderSection', () => {
+    return (
+        <header>
+            <div className="report-header-bar">
```
Comment (waabid) [inline src/electron/views/report/unified-header-section.tsx:11]: A couple things here:
1. I'd try to use css modules for all styles here.
2. I would also remove the ms-* styles as we should be clear about what styles we're using and why.

File: src/common/styles/fonts.scss [hunk 3]
```diff
@@ -79,6 +79,13 @@ $semiBoldFontFamily: make-font-family($semiBoldFonts) !default;
     line-height: 24px;
 }
 
+@mixin text-style-header-s {
```
Comment (waabid) [inline src/common/styles/fonts.scss:82]: the name should probably be headers instead of header-s

**Comment by waabid** [comment]
Could you include a screenshot for web as well?

File: src/common/styles/fonts.scss (hunk 3 above)
Comment (AhmedAbdoOrtiga) [reply src/common/styles/fonts.scss:82]: this style is mainly for the text font style, so I think it should be as it is, same for text-style-title-* and text-style-body-*
